### PR TITLE
SDPA Bwd Mapping

### PR DESCRIPTION
**Pull request by Priya2698**
This PR:

1. Removes `cum_seq_q/k` from `SdpaFwdOp` outputs and `SdpaBwdOp` inputs. These are undefined tensors for non-nested inputs which are not supported as fusion inputs. The nvFuser API now differs from torch/Thunder.
2. Adds pairwise domain mapping and scheduling support for `SdpaBwdOp`.
3. Allows tensor views marked as CPU scalars to be meta tensors. Detailed reasoning in code comments.

### CODE REVIEW

**Comment by Priya2698** [comment]
!build

**Comment by Priya2698** [comment]
!build

File: csrc/expr_evaluator.cpp [hunk 1]
```diff
@@ -55,10 +55,10 @@ void validateValWithConcreteValue(
         actual_dtype);
     if (tv->isCpuScalar()) {
       NVF_CHECK(
-          is_cpu_scalar(t),
+          is_cpu_scalar(t) || is_meta_scalar(t),
```
Comment (Priya2698) [inline csrc/expr_evaluator.cpp:58]: If an intermediate tensorview is CPU scalar, during compilation it is still created as a meta tensor since we do not yet have an actual tensor corresponding to that tensorview.

This occurs in fusions containing SDPA fwd and bwd ops.  The outputs from fwd ops -- `query_seq_len, key_seq_len, philox seed, philox offset` are CPU scalars.

If these are not marked as CPU scalars when writing the fusion, it will lead to an error in binding once the tensors have been computed. If they are marked as CPU scalars, then in the unmodified version, we run into an error in compilation since these intermediate tensors are created as meta tensors.

**Comment by Priya2698** [comment]
!build

**Comment by Priya2698** [comment]
!build

**Comment by Priya2698** [comment]
!build

File: csrc/expr_evaluator.cpp (hunk 1 above)
Comment (jacobhinkle) [reply csrc/expr_evaluator.cpp:58]: Can you summarize that in a comment at this location?

File: csrc/utils.cpp [hunk 2]
```diff
@@ -117,6 +117,10 @@ bool is_cpu_scalar(const c10::TensorType& tensor_type) {
       opt_numel.value() == 1;
 }
 
+bool is_meta_scalar(const at::Tensor& tensor) {
+  return tensor.device().is_meta() && tensor.numel() == 1 && tensor.dim() == 0;
```
Comment (jacobhinkle) [inline csrc/utils.cpp:121]: Is it not enough to check `tensor.device().is_meta()`?

File: csrc/utils.h [hunk 3]
```diff
@@ -60,6 +60,8 @@ bool is_zero_sized_tensor(const std::shared_ptr<c10::TensorType>& tensor_type);
 bool is_cpu_scalar(const at::Tensor& tensor);
 bool is_cpu_scalar(const c10::TensorType& tensor_type);
 
+bool is_meta_scalar(const at::Tensor& tensor);
```
Comment (jacobhinkle) [inline csrc/utils.h:63]: A short comment here might be helpful as it is not immediately obvious what a "meta scalar" is.

**Review by jacobhinkle — APPROVED** [review]
LGTM. Minor suggestions.

File: csrc/utils.cpp (hunk 2 above)
Comment (jjsjann123) [reply csrc/utils.cpp:121]: we do still want to verify that it's a scalar tensor, that's what the validation should do.

Comment (jacobhinkle) [reply csrc/utils.cpp:121]: Oh duh. Of course! Thanks.

File: csrc/utils.h (hunk 3 above)
Comment (jacobhinkle) [reply csrc/utils.h:63]: Feel free to ignore this, as it's essentially just a scalar `at::Tensor` that `.is_meta()` :-)
